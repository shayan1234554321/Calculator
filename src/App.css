:root {
  --orange: rgb(255, 185, 56);
}

* {
  font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
}

body {
  margin: 0;
  padding: 0;
}

.calContainer {
  display: flex;
  flex-wrap: wrap;
  width: 400px;
}

.display {
  padding: 5px;
  width: 390px;
  height: 40px;
  background-color: silver;
  color: white;
  display: flex;
  justify-content: end;
  align-items: center;
  font-size: 18px;
}

.buttons button {
  width: 100px;
  height: 50px;
  background-color: white;
  border: 1px solid rgba(0, 0, 0, 0.056);
  font-size: 20px;
  cursor: pointer;
  transition: 0.2s ease-in-out;
}

.buttons button:hover {
  filter: brightness(0.8);
}

.buttons button:nth-child(4n) {
  background-color: var(--orange);
}

.buttons button:nth-last-child(3) {
  width: 200px;
}

.buttons button:nth-last-child(1) {
  background-color: var(--orange);
}
