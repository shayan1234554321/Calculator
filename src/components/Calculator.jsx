import React from 'react'

export default function Calculator() {
  return (
    <div className='calContainer' >
        <div className="display">0</div>
        <form action="" className="buttons">
            <button>AC</button>
            <button>+/-</button>
            <button>%</button>
            <button>รท</button>
            <button>7</button>
            <button>8</button>
            <button>9</button>
            <button>x</button>
            <button>4</button>
            <button>5</button>
            <button>6</button>
            <button>-</button>
            <button>1</button>
            <button>2</button>
            <button>3</button>
            <button>+</button>
            <button>0</button>
            <button>.</button>
            <button>=</button>
        </form>
    </div>
  )
}
