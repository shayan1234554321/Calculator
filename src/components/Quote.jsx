import React, { useEffect } from 'react'
import ApiNinja from '../hooks/api'

export default function Quote() {

  const { data } = ApiNinja()

  useEffect(()=>{
    console.log(data)
  },[data])
  return (
    <div>
        {data? 
        <>
            Quote: {data[0].quote} <br />
            Author: {data[0].author} </>
        : "Loading Quote"}
    </div>
  )
}
