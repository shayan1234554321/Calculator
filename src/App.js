import './App.css';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import Calculator from './pages/calculator';
import Home from './pages/home';
import Quote from './pages/quote';
import Navbar from './components/navbar';

function App() {
  return (
    <Router>
      <Navbar />
      <Routes>
        <Route exact path="/" Component={Home} />
        <Route exact path="/calculator" Component={Calculator} />
        <Route exact path="/quote" Component={Quote} />
      </Routes>
    </Router>
  );
}

export default App;
