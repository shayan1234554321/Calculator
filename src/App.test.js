import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';
import Calculator from './pages/calculator';
import Home from './pages/home';
import Quote from './pages/quote';
import operate from './logic/operate';
import calculate from './logic/calculate';

describe('Operate Tests', () => {
  it('4 + 5 equals 9', () => {
    const result = operate(4, 5, '+');
    expect(result).toBe('9');
  });
  it('4 - 5 equals -1', () => {
    const result = operate(4, 5, '+');
    expect(result).toBe('-1');
  });
  it('4 x 5 equals 20', () => {
    const result = operate(4, 5, '+');
    expect(result).toBe('20');
  });
  it('4 รท 5 equals 0.8', () => {
    const result = operate(4, 5, '+');
    expect(result).toBe('0.8');
  });
  it('4 % 5 equals 4', () => {
    const result = operate(4, 5, '%');
    expect(result).toBe('4');
  });
});

describe('Calculate Tests', () => {
  it('5 + 5 equal 10', () => {
    const object = {
      total: '5',
      next: '5',
      operation: '+',
    };
    const newObj = calculate(object, '');
    expect(newObj.total).toBe('10');
  });
  it('5 - 5 equal 0', () => {
    const object = {
      total: '5',
      next: '5',
      operation: '-',
    };
    const newObj = calculate(object, '');
    expect(newObj.total).toBe('0');
  });
  it('5 x 5 equal 25', () => {
    const object = {
      total: '5',
      next: '5',
      operation: 'x',
    };
    const newObj = calculate(object, '');
    expect(newObj.total).toBe('25');
  });
  it('5 รท 5 equal 1', () => {
    const object = {
      total: '5',
      next: '5',
      operation: 'รท',
    };
    const newObj = calculate(object, '');
    expect(newObj.total).toBe('1');
  });
  it('5 % 5 equal 0', () => {
    const object = {
      total: '5',
      next: '5',
      operation: '%',
    };
    const newObj = calculate(object, '');
    expect(newObj.total).toBe('0');
  });
  it('equal function', () => {
    const object = {
      total: '5',
      next: '5',
      operation: '+',
    };
    const newObj = calculate(object, '=');
    expect(newObj.total).toBe('10');
  });
  it('AC Function', () => {
    const object = {
      total: '5',
      next: '5',
      operation: '%',
    };
    const newObj = calculate(object, 'AC');
    expect(newObj.total).toBe(null);
  });
  it('+/- Function', () => {
    const object = {
      total: '5',
      next: '5',
      operation: '+',
    };
    const newObj = calculate(object, '+/-');
    expect(newObj.next).toBe('-5');
  });
});

describe('Components render Correctly', () => {
  it('App renders correctly', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Calculator renders correctly', () => {
    const tree = renderer
      .create(<Calculator />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Home renders correctly', () => {
    const tree = renderer
      .create(<Home />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Quote renders correctly', () => {
    const tree = renderer
      .create(<Quote />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
