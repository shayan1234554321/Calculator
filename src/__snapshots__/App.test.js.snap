// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components render Correctly App renders correctly 1`] = `
Array [
  <nav>
    <div
      className="container"
    >
      <h1>
        Math Magicians
      </h1>
      <ul>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            HOME
          </a>
        </li>
        <li>
          <a
            href="/calculator"
            onClick={[Function]}
          >
            CALCULATOR
          </a>
        </li>
        <li>
          <a
            href="/quote"
            onClick={[Function]}
          >
            QUOTE
          </a>
        </li>
      </ul>
    </div>
  </nav>,
  <section
    className="home"
  >
    <div
      className="top"
    >
      <h2>
        Welcome to our page
      </h2>
      <p>
        Welcome to our website, where you can find a simple calculator to make your calculations easy and quick. Our calculator has a user-friendly interface that is easy to use, even if you're not a math expert. It can perform basic arithmetic operations like addition, subtraction, multiplication, and division, as well as more complex calculations like percentages, square roots, and exponents.
      </p>
    </div>
    <div
      className="bottom"
    >
      <p>
        In addition to our calculator, we also have a collection of inspiring quotes to brighten up your day. These quotes cover a variety of topics such as motivation, success, happiness, and more. We update our quotes regularly, so you can always find fresh inspiration whenever you visit our website.
      </p>
    </div>
  </section>,
]
`;

exports[`Components render Correctly Calculator renders correctly 1`] = `
<section
  className="calculator"
>
  <div
    className="calculatorCont"
  >
    <div
      className="calContainer"
    >
      <div
        className="display"
      >
        <input
          value="0"
        />
      </div>
      <div
        className="buttons"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <button
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
        <br />
        <button
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          x
        </button>
         
        <br />
        <button
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          -
        </button>
         
        <br />
        <button
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          +
        </button>
         
        <br />
        <button
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
  <div
    className="top"
  >
    <div
      className="extraCon"
    >
      <h3>
        Everyday
      </h3>
      <h2>
        CALCULATOR
      </h2>
    </div>
  </div>
  <div
    className="bottom"
  >
    <div
      className="extraCon"
    >
      <h4>
        Let's Do Some Math!
      </h4>
    </div>
  </div>
</section>
`;

exports[`Components render Correctly Home renders correctly 1`] = `
<section
  className="home"
>
  <div
    className="top"
  >
    <h2>
      Welcome to our page
    </h2>
    <p>
      Welcome to our website, where you can find a simple calculator to make your calculations easy and quick. Our calculator has a user-friendly interface that is easy to use, even if you're not a math expert. It can perform basic arithmetic operations like addition, subtraction, multiplication, and division, as well as more complex calculations like percentages, square roots, and exponents.
    </p>
  </div>
  <div
    className="bottom"
  >
    <p>
      In addition to our calculator, we also have a collection of inspiring quotes to brighten up your day. These quotes cover a variety of topics such as motivation, success, happiness, and more. We update our quotes regularly, so you can always find fresh inspiration whenever you visit our website.
    </p>
  </div>
</section>
`;

exports[`Components render Correctly Quote renders correctly 1`] = `
<section
  className="quote"
>
  <div
    className="top"
  >
    <p>
      "
      "-
    </p>
  </div>
  <div
    className="bottom"
  >
    <button
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Another Quote
    </button>
  </div>
</section>
`;

exports[`renders correctly 1`] = `
Array [
  <nav>
    <div
      className="container"
    >
      <h1>
        Math Magicians
      </h1>
      <ul>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            HOME
          </a>
        </li>
        <li>
          <a
            href="/calculator"
            onClick={[Function]}
          >
            CALCULATOR
          </a>
        </li>
        <li>
          <a
            href="/quote"
            onClick={[Function]}
          >
            QUOTE
          </a>
        </li>
      </ul>
    </div>
  </nav>,
  <section
    className="home"
  >
    <div
      className="top"
    >
      <h2>
        Welcome to our page
      </h2>
      <p>
        Welcome to our website, where you can find a simple calculator to make your calculations easy and quick. Our calculator has a user-friendly interface that is easy to use, even if you're not a math expert. It can perform basic arithmetic operations like addition, subtraction, multiplication, and division, as well as more complex calculations like percentages, square roots, and exponents.
      </p>
    </div>
    <div
      className="bottom"
    >
      <p>
        In addition to our calculator, we also have a collection of inspiring quotes to brighten up your day. These quotes cover a variety of topics such as motivation, success, happiness, and more. We update our quotes regularly, so you can always find fresh inspiration whenever you visit our website.
      </p>
    </div>
  </section>,
]
`;
